tools/c7n_azure/tests_azure/test_provider.py

```diff
@@ -27,3 +30,64 @@ def test_initialize_custom_account_id(self):
 
         session = azure.get_session_factory(options)()
         self.assertEqual(sample_account_id, session.get_subscription_id())
+
+    def test_initialize_default_azure_cloud(self):
+        with patch('c7n_azure.session.Session.get_subscription_id'):
+            options = Config.empty()
+            azure = Azure()
+            azure.initialize(options)
+            self.assertEqual(AZURE_PUBLIC_CLOUD, azure.cloud)
+            session = azure.get_session_factory(options)()
+
+        self.assertEqual(AZURE_PUBLIC_CLOUD.endpoints.resource_manager, session.base_url)
+
+    def test_initialize_default_azure_cloud_flag(self):
+        with patch('c7n_azure.session.Session.get_subscription_id'):
+            options = Config.empty(regions=[DEFAULT_AZURE_CLOUD])
+            azure = Azure()
+            azure.initialize(options)
+            self.assertEqual(AZURE_PUBLIC_CLOUD, azure.cloud)
+            session = azure.get_session_factory(options)()
+
+        self.assertEqual(AZURE_PUBLIC_CLOUD.endpoints.resource_manager, session.base_url)
+
+    def test_initialize_china_azure_cloud(self):
+        with patch('c7n_azure.session.Session.get_subscription_id'):
+            options = Config.empty(regions=['AzureChinaCloud'])

```

Comment:
are these values well known for azure cli? just wondering if we should take an opportunity to shorten, or if the values are well known (ie. strip the common cloud suffix, possibly other transforms).

Reply:
Running `az cloud list` returns the names: AzureCloud, AzureChinaCloud, AzureUSGovernment, and AzureGermanCloud as the names. I believe they are well known. Probably could change `USGov` -> `USGovernment` for consistency

Reply:
we should probably default to the well known vocabulary,  we're printing the value valid values on a bad one, so discoverability is good without resorting to a secondary source.